packages/components/src/sandbox/index.native.js

```diff
@@ -0,0 +1,252 @@
+		// determining the unconstrained page bounds.
+		function removeViewportStyles( ruleOrNode ) {
+			if( ruleOrNode.style ) {
+				[ 'width', 'height', 'minHeight', 'maxHeight' ].forEach( function( style ) {
+					if ( /^\\d+(vmin|vmax|vh|vw)$/.test( ruleOrNode.style[ style ] ) ) {
+						ruleOrNode.style[ style ] = '';
+					}
+				} );
+			}
+		}
+
+		Array.prototype.forEach.call( document.querySelectorAll( '[style]' ), removeViewportStyles );
+		Array.prototype.forEach.call( document.styleSheets, function( stylesheet ) {
+			Array.prototype.forEach.call( stylesheet.cssRules || stylesheet.rules, removeViewportStyles );
+		} );
+
+		document.body.style.position = 'absolute';
+		document.body.style.width = '100%';
+		document.body.setAttribute( 'data-resizable-iframe-connected', '' );
+
+		sendResize();
+
+		// Resize events can change the width of elements with 100% width, but we don't
+		// get an DOM mutations for that, so do the resize when the window is resized, too.
+		window.addEventListener( 'resize', sendResize, true );
+} )();`;
+
+const style = `
+	body {
+		margin: 0;
+	}
+	html,
+	body,
+	body > div,
+	body > div iframe {
+		width: 100%;
+	}
+	html.wp-has-aspect-ratio,
+	body.wp-has-aspect-ratio,
+	body.wp-has-aspect-ratio > div,
+	body.wp-has-aspect-ratio > div iframe {
+		height: 100%;
+		overflow: hidden; /* If it has an aspect ratio, it shouldn't scroll. */
+	}
+	body > div > * {
+		margin-top: 0 !important; /* Has to have !important to override inline styles. */
+		margin-bottom: 0 !important;
+	}
+`;
+
+function Sandbox( {
+	html = '',
+	providerUrl = '',
+	scripts = [],
+	styles = [],
+	title = '',
+	type,
+} ) {
+	const ref = useRef();
+	const [ width, setWidth ] = useState( 0 );
+	const [ height, setHeight ] = useState( 0 );
+	const [ contentHtml, setContentHtml ] = useState( getHtmlDoc() );
+
+	const windowSize = Dimensions.get( 'window' );
+	const [ isLandscape, setIsLandscape ] = useState(
+		windowSize.width >= windowSize.height
+	);
+	// On Android, we need to recreate the WebView when the device rotates, otherwise it disappears.
+	// For this purpose, the key value used in the WebView will change when the device orientation gets updated.
+	const key = Platform.select( {
+		android: `${ providerUrl }-${ isLandscape ? 'landscape' : 'portrait' }`,
+		ios: providerUrl,
+	} );
+
+	function getHtmlDoc() {
+		// TODO: Use the device's locale
+		const lang = 'en';
+
+		// Put the html snippet into a html document, and update the state to refresh the WebView,
+		// we can use this in the future to inject custom styles or scripts.
+		// Scripts go into the body rather than the head, to support embedded content such as Instagram
+		// that expect the scripts to be part of the body.
+		const htmlDoc = (
+			<html lang={ lang } className={ type }>
+				<head>
+					<title>{ title }</title>
+					<meta
+						name="viewport"
+						content="width=device-width, initial-scale=1"
+					></meta>
```

Comment:
This meta tag is used to adapt the content to the layout on mobile browsers ([more info about this tag here](https://developer.mozilla.org/en-US/docs/Web/HTML/Viewport_meta_tag)).

Reply:
Should we maybe add this on web version as well?

Reply:
Not really due to the way we render the preview in both native and web is a bit different. On the web version, the content is being rendered with an `iframe` component served within the editor page, which already contains the `viewport` metatag. On the native version, since we're using a WebView for each embed, we need to explicitly set the meta tag in the generated static HTML, as it serves as the page for rendering the preview.